libs/application/template-api-modules/src/lib/modules/templates/public-debt-payment-plan/public-debt-payment-plan.service.ts

```diff
@@ -59,35 +65,45 @@ export class PublicDebtPaymentPlanTemplateService {
   }
 
   async sendApplication({ application }: TemplateApiModuleActionProps) {
-    const { email, phoneNumber, paymentPlans } = this.getValuesFromApplication(
-      application,
-    )
+    try {
+      const {
+        email,
+        phoneNumber,
+        paymentPlans,
+      } = this.getValuesFromApplication(application)
 
-    const schedules = paymentPlans.map((plan) => {
-      const distribution = JSON.parse(plan.distribution) as PaymentsDT[]
-      return {
-        organizationID: plan.organization,
-        chargeTypes: plan.chargetypes?.map((chargeType) => ({
-          chargeID: chargeType.id,
-        })),
-        payments: distribution.map((p) => ({
-          duedate: p.dueDate,
-          payment: p.payment,
-          accumulated: p.accumulated,
-        })),
-        type: plan.id,
-      }
-    })
+      const schedules = paymentPlans.map((plan) => {
+        const distribution = JSON.parse(plan.distribution) as PaymentsDT[]
+        return {
+          organizationID: plan.organization,
+          chargeTypes: plan.chargetypes?.map((chargeType) => ({
+            chargeID: chargeType.id,
+          })),
+          payments: distribution.map((p) => ({
+            duedate: p.dueDate,
+            payment: p.payment,
+            accumulated: p.accumulated,
+          })),
+          type: ScheduleType[plan.id],
```

Comment:
Is it Ok, if `ScheduleType[plan.id],` returns undefined ?

Reply:
Yes since plan.id is actually of type PaymentScheduleType that maps to ScheduleType. If TBR (the external provider) adds a new schedule type we would have to update those types in our code. Plan.id comes from an external fetch to TBR's services, so plan.id should be of type PaymentScheduleType, otherwise something is wrong on their end.